src/test/java/com/stakater/nordmart/search/SearchApplicationTests.java

```diff
@@ -1,13 +1,96 @@
 package com.stakater.nordmart.search;
 
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.util.TestPropertyValues;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.config.KafkaListenerEndpointRegistry;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import pl.allegro.tech.embeddedelasticsearch.EmbeddedElastic;
+import pl.allegro.tech.embeddedelasticsearch.PopularProperties;
 
-@SpringBootTest
-class SearchApplicationTests {
+import java.util.concurrent.TimeUnit;
+
+@EnableKafka
+@SpringBootTest(
+		properties = "spring.kafka.bootstrap-servers=${spring.embedded.kafka.brokers}"
+)
+@EmbeddedKafka(
+		ports = {9092},
+		partitions = 1,
+		topics = { "productsTopic"})
+@ContextConfiguration(initializers = SearchApplicationTests.Initializer.class)
+public class SearchApplicationTests {
+	private static Integer port = 9200;
+	private static final String ELASTIC_VERSION = "6.5.4";
+	private static EmbeddedElastic embeddedElastic;
+	private static final String TEST_SERVER_URL = "http://localhost:8081";
+	private static final String CONTROLLER_PATH = "/v1/product-search";
+
```

Comment:
We should create a separate profile for  test and move most of these variables there.  "application-test.yml" that would just override the values like port & TEST_SERVER_URL

Reply:
Added